crates/ef-testing-proc-macro/src/utils.rs

```diff
@@ -0,0 +1,32 @@
+use std::path::{Path, PathBuf};
+
+use crate::dir_reader::PathWrapper;
+
+/// Converts the path to a string and removes the "BlockchainTests" folder
+pub(crate) fn blockchain_tests_to_general_state_tests_path(path: PathWrapper) -> PathWrapper {
+    Into::<PathBuf>::into(path)
+        .components()
+        .filter(|x| !x.as_os_str().eq_ignore_ascii_case("BlockchainTests"))
+        .collect::<PathBuf>()
+        .into()
+}
+
+/// Converts the path to a Vector of Strings
+pub(crate) fn path_to_vec_string(path: &Path) -> Result<Vec<String>, eyre::Error> {
```

Comment:
why do we need to go from having Paths to having vectors of strings?

Reply:
The issue I ran into was that it was a lot easier to handle a Vec<String> over a Path: using a Vec<String>, I can pop the first value in the array, store that in my `DirReader` structure and keep recursing. However, a `Path` doesn't have a pop_front method, and I was stuck with some complicated helpers. I found the method of just using `Vec<String>` easier.

Reply:
I'm surprised there is no known library to recursively explore a directory and create a sort of tree for all files inside it

Reply:
You can walk a directory but I was not able to find a known crate that allows you to recursively construct a mapping of the folders.

Reply:
Do we need to go from path to vec<string> to accomplish our goal (which is to recursively find all tests?)
could we make it work with:
https://github.com/paradigmxyz/reth/blob/main/testing/ef-tests/src/suite.rs#L50

Reply:
I already use Walkdir to find all the file, this doesn't really fix the main issue, which is that I wasn't able to find an easy way to just `pop_front` the first folder in a `Pathbuf`. I did use a solution with Pathbuf at the beginning but using a `Vec<String>` removed 50% of the code, which was mainly doing `.iter().take(1).collect()` everywhere and also `iter().skip(1).collect()`.